proto/api/v1/api.proto

```diff
@@ -799,3 +800,61 @@ option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
     }
   }
 };
+
+service PluginService {
+  rpc ListPlugins(ListPluginsRequest) returns (ListPluginsResponse) {
+    option (google.api.http) = {
+      get: "/v1/plugins"
+      response_body: "plugins"
+    };
+  };
+  rpc GetPlugin(GetPluginRequest) returns (GetPluginResponse) {
+    option (google.api.http) = {
+      get: "/v1/plugin/{name}"
```

Comment:
I'm thinking about the name, it can either be `builtin:[plugin-name]` or the path to the plugin. The path might be absolute (i.e. start with a slash), I doubt the grpc-gateway would even parse this correctly. Sidenote - this also made me think about #180, where we need an endpoint for validating a plugin config, in that case something like `/v1/plugin/{name}/validate` could make sense, but it wouldn't be possible to determine if the `/validate` is part of the name or the action. We might have to rethink how we identify plugins.

Anyway, we could remove this endpoint and instead add a name filter in the `ListPlugins` rpc, same as we do in `ListPipelines`, I think that makes the most sense right now.